File: backend/models/StudySession.js
const mongoose = require('mongoose');

const studySessionSchema = new mongoose.Schema({
  courseId:       { type: String, required: true },
  organizerId:   { type: String, required: true },
  organizerName: { type: String, required: true },
  title:         { type: String, required: true },
  description:   { type: String, default: '' },
  location:      { type: String, default: '' },
  dateTime:      { type: Date, required: true },
  duration:      { type: Number, default: 60 }, // minutes
  attendingIds:     [{ type: String }],
  notAttendingIds:  [{ type: String }],
}, { timestamps: true });

studySessionSchema.methods.toJSON = function() {
  const obj = this.toObject();
  obj.id = obj._id.toString();
  delete obj._id;
  delete obj.__v;
  return obj;
};

studySessionSchema.index({ courseId: 1 });

module.exports = mongoose.model('StudySession', studySessionSchema);

File: lib/models/study_session.dart
class StudySession {
  final String id;
  final String courseId;
  final String organizerId;
  final String organizerName;
  final String title;
  final String description;
  final String location;
  final DateTime dateTime;
  final int duration; // in minutes
  final List<String> attendingIds;
  final List<String> notAttendingIds;

  const StudySession({
    required this.id,
    required this.courseId,
    required this.organizerId,
    required this.organizerName,
    required this.title,
    required this.description,
    required this.location,
    required this.dateTime,
    required this.duration,
    this.attendingIds = const [],
    this.notAttendingIds = const [],
  });

  factory StudySession.fromJson(Map<String, dynamic> json) {
    return StudySession(
      id: json['id'] ?? json['_id'] ?? '',
      courseId: json['courseId'] ?? '',
      organizerId: json['organizerId'] ?? '',
      organizerName: json['organizerName'] ?? '',
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      location: json['location'] ?? '',
      dateTime: DateTime.tryParse(json['dateTime'] ?? '') ?? DateTime.now(),
      duration: json['duration'] ?? 60,
      attendingIds: List<String>.from(json['attendingIds'] ?? []),
      notAttendingIds: List<String>.from(json['notAttendingIds'] ?? []),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'courseId': courseId,
    'organizerId': organizerId,
    'organizerName': organizerName,
    'title': title,
    'description': description,
    'location': location,
    'dateTime': dateTime.toIso8601String(),
    'duration': duration,
    'attendingIds': attendingIds,
    'notAttendingIds': notAttendingIds,
  };
}
